doc/game_design.md

```diff
@@ -0,0 +1,48 @@
+# Game Design
+Drakkar is a multiplayer game featuring dragons of different colours trying to chase and eat one another.
+The game resembles [agar.io](https://agar.io/) with the addition of a Rock-Paper-Scissor dynamic via the RGB colours.
+
+The goals and rules of the game are described here. We aim to make the design described in this doc come to life, and should refrain from working on things that do not fall in its scope. Any new feature or idea should be discussed first, then if received positively should be included in the design doc. 
+
+## General Principle
+There are three players. Each player controls a disc-shaped dragon of one of three colours: red, green, and blue.
+Every colour is strong against another one and weak against the third one. 
+Players chase one another, trying to run into and eat the weaker player while evading the stronger player.
+
+### Interactions
+A player running into another one triggers a collision. 
+Upon a collision, the colour of the involved dragons is compared, and the dragon with the strongest colour wins, according to the following Rock-Paper-Scissor logic:
+
+- Red is strong against Green, and weak against Blue;
+- Green is strong against Blue, and weak against Red;
+- Blue is strong against Red, and weak against Green.
+
+The winning dragon then munches on the losing one, growing a bit in the process, while the loser shrinks a bit.
+The loser then turns invulnerable for a short duration, so that it has a chance to escape its victor and avoid being munched on indefinitely.
+During this time, the invulnerable dragon cannot be involved in a collision with the stronger player (it simply ignores it as it runs into it), but can still collide with the weaker player.
+
+### Winning condition
+The game is timed; when the timer runs out, whichever player has the largest dragon wins the game.
+In the event of a tie, the winner is decided on a coin flip among the tied players.
+
+### Death and size limitations
+As described above, players change size upon interaction with other players.
+When a player falls under a certain size, it "dies". 
+While dead, the player cannot interact with either of the other players (and so ignores collisions).
+After some time, the player is revived, upon which it returns to its starting size and can interact with other players again.
```

Comment:
I am curious about the fairness to return to the starting size after revival, but it is hard to say and I don't have experience playing arga.io, we will see after achieving the minimal viable game of course. 

Also, do the kill/death stats matter in determining the winner? Might be useful if there is a tie.

Reply:
> Also, do the kill/death stats matter in determining the winner? Might be useful if there is a tie.

They don't, so far we solve ties with a coin flip. I like the idea of using death stats instead, sounds more fair! Let's keep it in the back of our heads for now.

Reply:
> I am curious about the fairness to return to the starting size after revival, but it is hard to say and I don't have experience playing arga.io, we will see after achieving the minimal viable game of course.

This is because of the "natural decay" thing. If a player reaches the minimum size and dies, then it has no way to interact and thus it cannot grow. Upon reviving, because of the decay it will immediately die again. Being revived at your starting size avoids this. Is there a logic flaw in this?